assets/js/googlesitekit/datastore/site/notifications.test.js

```diff
@@ -21,18 +21,223 @@
-			expect.arrayContaining( actionsToExpect )
-		);
+	let registry;
+
+	const markNotificationsEndpoint = /^\/google-site-kit\/v1\/core\/site\/data\/mark-notification/;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'fetchMarkNotification', () => {
+			it.each( [
+				[ 'abc', 'accepted' ],
+				[ 'abc', 'dismissed' ],
+			] )(
+				'should not throw if the notificationID is "%s" and notification state is "%s".',
+				( notificationID, notificationState ) => {
+					fetchMock.postOnce( markNotificationsEndpoint, {
+						body: 'true',
+						status: 200,
+					} );
+					expect( () =>
+						registry.dispatch( CORE_SITE ).fetchMarkNotification( {
+							notificationID,
+							notificationState,
+						} )
+					).not.toThrow();
+					expect( fetchMock ).toHaveFetched(
+						markNotificationsEndpoint,
+						{
+							body: {
+								data: {
+									notificationID,
+									notificationState,
+								},
+							},
+						}
+					);
+				}
+			);
+			it.each( [
+				[ 'abc', 'test' ],
+				[ 'abc', undefined ],
+				[ undefined, 'accepted' ],
+			] )(
+				'throws an error while trying to marks a notification when the notification ID is "%s" and notification state is "%s".',
+				( notificationID, notificationState ) => {
+					expect( () =>
+						registry.dispatch( CORE_SITE ).fetchMarkNotification( {
+							notificationID,
+							notificationState,
+						} )
+					).toThrow();
+				}
+			);
+		} );
+		describe( 'acceptNotification', () => {
+			it( 'accepts a notification with a valid notification ID.', () => {
+				fetchMock.postOnce( markNotificationsEndpoint, {
+					body: 'true',
+					status: 200,
+				} );
+				expect( () =>
+					registry.dispatch( CORE_SITE ).acceptNotification( 'abc' )
+				).not.toThrow();
+				expect( fetchMock ).toHaveFetched( markNotificationsEndpoint, {
+					body: {
+						data: {
+							notificationID: 'abc',
+							notificationState: 'accepted',
+						},
+					},
+				} );
+			} );
+
+			it( 'sets an error when API returns one.', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetchMock.postOnce( markNotificationsEndpoint, {
+					body: response,
+					status: 500,
+				} );
+				await registry
+					.dispatch( CORE_SITE )
+					.acceptNotification( 'abc' );
+				expect(
+					registry
+						.select( CORE_SITE )
+						.getErrorForAction( 'acceptNotification', [ 'abc' ] )
+				).toMatchObject( response );
+				expect( fetchMock ).toHaveFetched( markNotificationsEndpoint, {
+					body: {
+						data: {
+							notificationID: 'abc',
+							notificationState: 'accepted',
+						},
+					},
+				} );
+				expect( console ).toHaveErrored();
+			} );
+			it( 'sets an error when request throws an error while trying to accept a notification.', () => {
```

Comment:
This test repeats the first test for the `acceptNotification` action. Do we need this? Looks like we can just delete it, can't we?

Reply:
https://github.com/google/site-kit-wp/pull/4349/commits/cc65bbcc9df2e50685ecdbd55804b39e0eac56c9